lib/gui/components/update-notifier/services/update-notifier.js

```diff
@@ -60,7 +60,9 @@ module.exports = function($http, $q, ModalService, UPDATE_NOTIFIER_SLEEP_TIME, M
           // The error status equals -1 if the request couldn't
           // be made successfully, for example, because of a
           // timeout on an unstable network connection.
-          if (error.status === -1) {
+          const ERROR_CODE_UNSUCCESSFULL_REQUEST = -1;
```

Comment:
Hm, I investigated this one in more detail and it turns out its a bit tricky. The originator of this `-1` error code when the request couldn't be made is `$http`. `etcherLatestVersion` accepts a  function that given a URL allows the client to determine how to perform an HTTP request to it (given we use the module in the browser as well).

Since `etcherLatestVersion` accepts a function to make HTTP requests, then hardcoding the `-1` knowledge there looks a bit wrong, since different HTTP request functions might use different error codes for different things.

That leaves us with a few options:

- Expose a set of error constants from `etcherLatestVersion` and update the `.catch` block of the `$http` promise to compare against `-1` and translate that to an error number `etcherLatestVersion` understands, but that means we still have to declare this `-1` constant here

- Make `etcherLatestVersion` ignore every error coming from the HTTP requester function. This means we will swallow errors that we should be seeing

- Leave it as it is now

What do you think?